probe-rs/src/probe.rs

```diff
@@ -676,21 +679,29 @@ pub trait JTAGAccess: DebugProbe {
         len: u32,
     ) -> Result<Vec<u8>, DebugProbeError>;
 
-    /// Schedules a register write to be executed later by calling `execute`
-    ///
-    /// Returns a DeferredCommandResult which can be used to get the result after `execute`.
-    /// If the probe doesn't support batching this will be emulated.
-    fn schedule_write_register(
+    fn write_register_batch(
         &mut self,
-        address: u32,
-        data: &[u8],
-        len: u32,
-        transform: fn(Vec<u8>) -> Result<u32, DebugProbeError>,
-    ) -> Result<Box<dyn DeferredCommandResult>, DebugProbeError>;
+        writes: &[JtagWriteCommand],
+    ) -> Result<Box<dyn CommandResults>, BatchExecutionError> {
```

Comment:
It seems like the `CommandResults` trait is no longer needed? Unless I'm missing something, this could be simplified to

```rust
fn write_register_batch(
        &mut self,
        writes: &[JtagWriteCommand],
    ) -> Result<Vec<CommandResult>, BatchExecutionError> {
```